modules/service/src/main/scala/lucuma/odb/graphql/topic/DatasetTopic.scala

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql.topic
+
+import cats.effect.Concurrent
+import cats.effect.std.Supervisor
+import cats.syntax.all.*
+import fs2.concurrent.Topic
+import lucuma.core.model.Observation
+import lucuma.core.model.Program
+import lucuma.core.model.User
+import lucuma.core.model.sequence.Dataset
+import lucuma.core.util.Gid
+import lucuma.odb.data.EditType
+import org.typelevel.log4cats.Logger
+import skunk.*
+import skunk.syntax.stringcontext.*
+
+object DatasetTopic:
+
+  case class Element(
+    datasetId:     Dataset.Id,
+    observationId: Observation.Id,
+    programId:     Program.Id,
+    isWritten:     Boolean,
+    editType:      EditType,
+    users:         List[User.Id]
+  ) extends TopicElement
+
+  private val topic =
+    OdbTopic.define[(Dataset.Id, Observation.Id, Program.Id, Boolean, EditType), Element](
+      "Dataset",
+      id"ch_dataset_edit",
+      _._3,
+      (update, users) => Element(update._1, update._2, update._3, update._4, update._5, users)
+    ) {
+      case Array(_did, _oid, _pid, _isWritten, _tg_op) =>
+        (
+          Gid[Dataset.Id].fromString.getOption(_did),
+          Gid[Observation.Id].fromString.getOption(_oid),
+          Gid[Program.Id].fromString.getOption(_pid),
+          "true".equalsIgnoreCase(_isWritten).some,
```

Comment:
Why is this hardcoded?

Reply:
Here is the corresponding `pg_notify` that produces the data

```plpgsql
  PERFORM pg_notify(
    'ch_dataset_edit',
    array_to_string(
      ARRAY[
        NEW.c_dataset_id,
        NEW.c_observation_id,
        pid,
        -- isWritten
        CASE WHEN NEW.c_end_time IS NOT NULL THEN 'true' ELSE 'false' END,
        TG_OP
      ],
      ','
    )
  );
```

We need to parse the `String` array elements into the types we expect.  It's either `true` or `false` according to the notify but I just treat everything other than `"true"` as `false`.  There's `java.lang.Boolean.parseBoolean` that I could probably use instead.  Or probably there's a better option?

Reply:
ah right, seems fine